syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package d_auth;


message AKAResyncInfo {
    bytes rand = 1;
    bytes auts = 2;
}

// The 5G-AKA authentication vector, explicitly with a hashed xres.
message AuthVector5G {
    // The random seed defined for this vector
    bytes rand = 1;

    // The sha256 hashed expected response from the UE
    //
    // TS33.501 Annex A.5
    bytes xres_star_hash = 2;

    // The autn value to pass to the UE, containing the network AUTH and SQN
    bytes autn = 3;
}

// The type of id provided.
enum UserIdKind {
    UNKNOWN = 0;
    SUPI = 1;
}
